src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TimelineEventFactory.kt

```diff
@@ -67,59 +85,73 @@ class TimelineEventFactory {
       )
     }
 
-    if (hasGoalStatusChanged(previousGoal = previousGoal, updatedGoal = updatedGoal)) {
-      val eventType = getGoalStatusEventType(updatedGoal)
-      events.add(
+    // check if any steps have been changed
+    val stepEvents = getStepUpdatedEvents(updatedGoal, previousGoal, correlationId)
+
+    // if one or more Steps have changed, but the Goal itself hasn't been modified, then record an overall GOAL_UPDATED event
+    if (timelineEvents.isEmpty() && stepEvents.isNotEmpty()) {
```

Comment:
I thought about creating a private method for this if statement, but I didn't know what to call it! If `stepsHaveBeenModifiedButTheParentGoalItselfHasn'tBeen()` 😂